src/IECoreGL/TextureLoader.cpp

```diff
@@ -37,19 +37,27 @@
 #include "IECoreGL/Texture.h"
 #include "IECoreGL/ToGLTextureConverter.h"
 
+#include "IECoreImage/ImageReader.h"
+
+#include "IECore/NumericParameter.h"
 #include "IECore/MessageHandler.h"
 #include "IECore/Reader.h"
 
+#include <limits>
+
 using namespace IECoreGL;
 
 TextureLoader::TextureLoader( const IECore::SearchPath &searchPaths )
 	:	m_searchPaths( searchPaths )
 {
 }
 
-TexturePtr TextureLoader::load( const std::string &name )
+TexturePtr TextureLoader::load( const std::string &name, const IECore::IntData *maximumResolution )
 {
-	TexturesMap::const_iterator it = m_loadedTextures.find( name );
+	int dataWindowSizeLimit = maximumResolution ? maximumResolution->readable() : std::numeric_limits<int>::max();
+
+	TexturesMapKey key = std::make_pair( name, dataWindowSizeLimit );
```

Comment:
As I understand it, the main purpose of `make_pair` is to avoid typing out the types, and for generic code where determinging the type might be a pain. Here `TexturesMapKey key( name, dataWindowSizeLimit )` is shorter and clearer.

Reply:
Fixed as part of 2ae7040. Thanks, you're completely right, of course.